### PR TITLE
Fix #3037: Add ResetPedVoice function - Bug fix

### LINKED ISSUE #3037: Add resetPedVoice function
### Is your feature request related to a problem? Please describe.

we have now
setPedVoice
and
getPedVoice

... i wanted to make a ped mute ( even in pain he doesn't say much )
so i did setPedVoice( "disabled bank name" )

and it worked, the ped is perfectly silent... problem is when i want to undo this, at first i expected that changing my skin to another skin would override this and restore my voice, but it didn't..

ped remained mute until rejoin

i did think about using getPedVoice to store a backup of what i had before muting... but then when i changed the skin to a female and was previously male... the backup values no longer matched !

### Describe the solution you'd like

i would like resetPedVoice to be introduced as a native function that resets the voice bank to your skin IDs' native original bank !

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### Security Policy

- [X] I have read and understood the Security Policy and this issue is not about a cheat or security vulnerability.

### PR DESCRIPTION
**Pull request by TracerDS**
The function was bugged; it was never actually resetting ped's voice due to internal mta's logic.
Now should be fixed, function returns `true` when the voice was successfully reset, returns `false` if the voice was not changed (ie. player joins the server, player executes `resetPedVoice` without ever calling `setPedVoice`)

The old PR (#3042) was merged, so I had to make new PR

### CODE REVIEW

File: Client/game_sa/CPedSA.h [hunk 1]
```diff
@@ -265,24 +264,24 @@ class CPedSA : public virtual CPed, public virtual CPhysicalSA
 
 private:
     CWeaponSA*          m_pWeapons[WEAPONSLOT_MAX];
-    CPedIKSA*           m_pPedIK;
-    CPedIntelligenceSA* m_pPedIntelligence;
-    CPedSAInterface*    m_pPedInterface;
-    CPedSoundSA*        m_pPedSound;
+    CPedIKSA*           m_pPedIK = NULL;
```
Comment (sbx320) [inline Client/game_sa/CPedSA.h]: nullptr

**Review by sbx320 — CHANGES_REQUESTED** [review]

File: Client/game_sa/CPedSoundSA.h [hunk 2]
```diff
@@ -70,15 +70,22 @@ class CPedSoundSA : public CPedSound
     CPedSoundSAInterface* m_pInterface;
 
 public:
+    CPedSoundSA(CPedSoundSAInterface&& pInterface) noexcept
```
Comment (tederis) [inline Client/game_sa/CPedSoundSA.h]: What is the purpose of rvalue reference here?

File: Client/game_sa/CPedSA.cpp [hunk 3]
```diff
@@ -94,7 +88,7 @@ void CPedSA::Init()
     CPedIntelligenceSAInterface* m_pPedIntelligenceInterface = (CPedIntelligenceSAInterface*)(dwPedIntelligence);
     m_pPedIntelligence = new CPedIntelligenceSA(m_pPedIntelligenceInterface, this);
     m_pPedSound = new CPedSoundSA(&pedInterface->pedSound);
-    m_pDefaultPedSound = new CPedSoundSA(&pedInterface->pedSound);
+    m_pDefaultPedSound = new CPedSoundSA(std::move(pedInterface->pedSound));
```
Comment (tederis) [inline Client/game_sa/CPedSA.cpp]: What exactly are you trying to move here? CPedSoundSA constructor below doesn't implement any move.

File: Client/game_sa/CPedSoundSA.h (hunk 2 above)
Comment (TracerDS) [reply Client/game_sa/CPedSoundSA.h]: to distinguish it from the normal sound

File: Client/game_sa/CPedSA.cpp (hunk 3 above)
Comment (TracerDS) [reply Client/game_sa/CPedSA.cpp]: `std::move` transforms pedSound into a rvalue. pedSound need to be a copy and by `std::move`ing it, it kinda achieves it but I could've better written it now that I think of it

File: Client/game_sa/CPedSoundSA.h (hunk 2 above)
Comment (tederis) [reply Client/game_sa/CPedSoundSA.h]: Nonsence! When you're using a rvalue reference you should use it as a rvalue reference. 

```
CPedSoundSA(CPedSoundSAInterface* pInterface, bool bInstantiate = false) :
        m_pInterface(bInstantiate ? new CPedSoundSAInterface() : pInterface)
{
        if (bInstantiate)
        {
                SetVoiceID(pInterface->m_sVoiceID);
                SetVoiceTypeID(pInterface->m_sVoiceType);
        }
}
```
would be much better

File: Client/game_sa/CPedSA.cpp (hunk 3 above)
Comment (tederis) [reply Client/game_sa/CPedSA.cpp]: This is counterintuitive and breaks principles of the C++ standard.

**Pull request merged**